### PR TITLE
ci: make e2e test dependant

### PR DESCRIPTION
**Pull request by petertonysmith94**
- E2E tests now dependant on other tests
- E2E tests only run for non-draft PR's
- Dependabot will run weekly (instead of daily)

### CODE REVIEW

**Review by nedsalk — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Review by Torres-ssf — COMMENTED** [review]
@petertonysmith94 maybe we could also make the`e2e` job run only if the PR is marked as ready for review.

**Comment by Torres-ssf** [comment]
> > @petertonysmith94 maybe we could also make the`e2e` job run only if the PR is marked as ready for review.
> 
> I like your thinking @Torres-ssf! Would this be a combination of the current behaviour and ready for review? Or just ready for review?

@petertonysmith94 Yes, I believe a combination of these 2 validations would be the best way to avoid unnecessary spending on our tokens.

**Review by maschad — CHANGES_REQUESTED** [review]
As mentioned in our slack convo, I think another useful addition would be to change the dependabot [interval](https://github.com/FuelLabs/fuels-ts/issues/2162) to weekly  or even monthly https://github.com/FuelLabs/fuels-ts/blob/master/.github/dependabot.yml#L6 in tandem with the other changes

**Review by Torres-ssf — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

File: .changeset/shaggy-cougars-invent.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
[the remark this answers is not in the document]
Comment (maschad) [reply .changeset/shaggy-cougars-invent.md:2]: I think it can be removed.

**Review by maschad — APPROVED** [review]
Nice

File: .changeset/shaggy-cougars-invent.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply .changeset/shaggy-cougars-invent.md:2]: resolved: https://github.com/FuelLabs/fuels-ts/commit/ec2d02f0564a473bf28e0e21772c7946b53059a7

**Pull request merged**